### PR TITLE
Feature/sb6

### LINKED ISSUE #108: Storybook 6 support by default
👋  A snag list for things that I noticed when setting the required version to `^6.0.0` when invoking the plugin. I'll do my best to submit PRs for all of these

- [x] The glob is now incorrect. It should have an `@` before the `(js|jsx....)`
- [x] Instead of installing all the dependencies, it can use the `@storybook/essentials` package
- [x] This will enable the `main.js` file to be simpler, and it won't need the Vue/JSX hack
- [x] The default stories should use `<canvas>` instead of `<preview>`
- [x] Add documentation to `MyButton.vue` so that it can show off auto-documentation in `<ArgsTable>`

### PR DESCRIPTION
**Pull request by steveworkman**
Resolves #108

### CODE REVIEW

File: generator/template/config/storybook/main.js [hunk 1]
```diff
@@ -29,3 +29,8 @@ module.exports = {
   ]
 }
 <%_ } _%>
+<%_ if (is_6_0) { _%>
+  module.exports = {
+    stories: ['../../src/**/*.stories.@(js|jsx|ts|tsx|mdx)']
+  }
+  <%_ } _%>
```
Comment (pksunkara) [inline generator/template/config/storybook/main.js:36]:
```suggestion
<%_ } _%>
```

Comment (pksunkara) [reply generator/template/config/storybook/main.js:36]: Also `\n` at end of file.

**Review by pksunkara — CHANGES_REQUESTED** [review]
Just minor thing left. Also, please remove the merge commit.

> The default stories should use <canvas> instead of <preview>
> Add documentation to MyButton.vue so that it can show off auto-documentation in <ArgsTable>

These two items doesn't seem to be done though, but I don't particularly mind them.

**Comment by pksunkara** [comment]
Also how does this interact with #110?

**Comment by steveworkman** [comment]
- This is a draft PR - it's not ready yet. Not all the features in the checklist are there. The feedback is great though
- Will remove the merge
- I wasn't aware of #110 - happy to add

**Review by pksunkara — COMMENTED** [review]
Since you updated the test script, I am okay with removing support for 5.x and we can do a major version.

**Comment by steveworkman** [comment]
@pksunkara ok - I'll strip out all the old code and do a major bump

**Comment by steveworkman** [comment]
Circle CI timed out connecting to yarn, and I don't have permission to re-run it. Can you do that @pksunkara ?

File: generator/template/src/stories/index.stories.js [hunk 2]
```diff
@@ -1,39 +1,12 @@
 <%_ if (hasEslintPluginImport) { _%>
 /* eslint-disable import/no-extraneous-dependencies */
 <%_ } _%>
-<%_ if (!csf) { _%>
-import { storiesOf } from '@storybook/vue'
-<%_ } _%>
 import { action } from '@storybook/addon-actions'
-<%_ if (hasBabel) { _%>
```
Comment (pksunkara) [inline generator/template/src/stories/index.stories.js:8]: Should not be removed.

File: generator/template/src/stories/index.stories.js [hunk 3]
```diff
@@ -42,17 +15,14 @@ export const withText = () => ({
   template: '<my-button @click="action">Hello Button</my-button>',
   methods: { action: action('clicked') }
 })
-<%_ if (hasBabel) { _%>
```
Comment (pksunkara) [inline generator/template/src/stories/index.stories.js:45]: Should not be removed.

File: generator/template/src/stories/index.stories.js [hunk 4]
```diff
@@ -1,39 +1,12 @@
...
-  }))
-  <%_ } _%>
-  .add('With Some Emoji', () => ({
-    components: { MyButton },
-    template: '<my-button>😀 😎 👍 💯</my-button>'
-  }))
-<%_ } else { _%>
 export default {
-  <%_ if (is_5_3) { _%>
-  component: MyButton,
```
Comment (pksunkara) [inline generator/template/src/stories/index.stories.js:35]: This line should not be removed.

File: prompts.js [hunk 5]
```diff
@@ -1,48 +0,0 @@
-const semver = require('semver');
```
Comment (pksunkara) [inline prompts.js:1]: Why is this file removed? Just remove the prompts that are removed. Don't remove the whole file.

**Review by pksunkara — CHANGES_REQUESTED** [review]

File: generator/template/src/stories/index.stories.js (hunk 2 above)
Comment (steveworkman) [reply generator/template/src/stories/index.stories.js:8]: Why? does the `addon-links` plugin not function without babel?

File: generator/template/src/stories/index.stories.js (hunk 3 above)
Comment (steveworkman) [reply generator/template/src/stories/index.stories.js:45]: Re-added

File: prompts.js (hunk 5 above)
Comment (steveworkman) [reply prompts.js:1]: Once the version number check has been removed (as we support v6 only) then the prompts file becomes redundant:

- Type only ever supported "init" - no need for a prompt if there's only one option
- semver check has been removed
- docs option is redundant as it comes as part of essentials
- csf is the default way to write stories, and you really shouldn't be turning it off in the options

I'd like to think that these should be a best-practice starting point, showing developers how things can be done. They're free to delete these template files when they don't need them. With 6.0, docs and csf on or off would have no effect on any dependencies being installed - just what templates are provided

File: generator/template/src/stories/index.stories.js (hunk 2 above)
Comment (pksunkara) [reply generator/template/src/stories/index.stories.js:8]: No, but we show jsx story if babel is enabled. And it uses addon links. So, this section is enabled only when babel is enabled

File: prompts.js (hunk 5 above)
Comment (pksunkara) [reply prompts.js:1]: Yeah, but I want to do different types of generations. So, let's keep the type prompt

File: generator/template/src/stories/index.stories.js (hunk 2 above)
Comment (steveworkman) [reply generator/template/src/stories/index.stories.js:8]: Understood. My bad

File: prompts.js (hunk 5 above)
Comment (steveworkman) [reply prompts.js:1]: ok

File: prompts.js [hunk 6]
```diff
@@ -13,36 +10,4 @@ module.exports = [
...
-        return 'Given semver range is not valid.';
-      }
-
-      if (!semver.prerelease(input) && !semver.intersects(input, '>=4.1.0')) {
-        return 'Minimum supported storybook version is 4.1.0';
-      }
-
-      return true;
-    },
-  },
```
Comment (pksunkara) [inline prompts.js:33]: I think we should keep this storybook version question and change the minimum storybook version to 6

Comment (steveworkman) [reply prompts.js:33]: Done

**Review by pksunkara — APPROVED** [review]
